src/sentry/api/endpoints/organization_member_index.py

```diff
@@ -43,3 +72,89 @@ def get(self, request, organization):
             on_results=lambda x: serialize(x, request.user),
             paginator_cls=OffsetPaginator,
         )
+
+    def post(self, request, organization):
+        """
+        Add a Member to Organization
+        ````````````````````````````
+
+        Invite a member to the organization.
+
+        :pparam string organization_slug: the slug of the organization the member will belong to
+        :param string email: the email address to invite
+        :param string role: the role the new member
+        :param array teams: the slugs of the teams the member should belong to.
+
+        :auth: required
+        """
+        # TODO: If the member already exists, should this still update the role and team?
+        # For now, it doesn't, but simply returns the existing object
+
+        serializer = OrganizationMemberSerializer(data=request.DATA)
+
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=400)
+
+        result = serializer.object
+
+        can_admin, allowed_roles = get_allowed_roles(request, organization)
+
+        if not result['role'] in {r.id for r in allowed_roles}:
+            return Response({'role': 'You do not have permission to invite that role.'}, 403)
+
```

Comment:
another thing you should probably validate here is that the teams provided actually exist. You could do something like:
```
teams = list(Team.objects.filter(
    organization=organization,
    status=TeamStatus.VISIBLE,
    slug__in=result['teams'],
))
if len(set(result['teams'])) != len(teams):
    return Response({'teams': 'Invalid team', 400})
```
and then you can reuse that list of teams below for the `save_team_assignments`